packages/slides-frontend/src/containers/redux-store/DeckReducer/reducer.ts

```diff
@@ -47,6 +84,37 @@ const DeckState = (state: Deck = initialDeck, action: Action): Deck =>
         } else alert('Not possible to remove the only slide');
         break;
       }
+      case CLONE_SLIDE: {
+        // create a new blank slide
+        let slide:Slide = {
+          ID: uuidv4(),
+          itemsArray: [],
+        }
+        // with a loop over the current itemsArray, create and push newItems in the new itemsArray
+        slide.itemsArray = draft.slides[draft.currentSlide].itemsArray.map(currentItem => {
+          const copyObj = JSON.parse(JSON.stringify(currentItem));
+          if(copyObj.type === ItemTypes.TEXT) {
+            const itemObj: Text = new Text();
+            itemObj.Data = copyObj.Data;
+            // change size and position
+            itemObj.Size = copyObj.Size;
+            itemObj.Position = copyObj.Position;
+            return itemObj;
+          }
+
+          // if it is IMAGE
+          const { Src } = copyObj;
```

Comment:
`clone` method on those classes would make it much more generic and extensible.

Reply:
Same as above.
Is the best approach to create the two clone methods?

Reply:
the best would be to have an interface/base class of all those classes .e.g. `Clonable` that would define a single method `clone`. Then it would be up to a specific inheriting class how this method would be implemented

Reply:
So the order of inheritance should be:
* Clonable
    * baseItem
        * Text
        * Image
and the clone method: 
``` js
public clone():Clonable {
    let copy = new (this.constructor());
    copy = deepCopyFunction(this)
    return copy;
  }
```

and then the itemsArray contains Items, which is
`export type Item = Text | Image;`
but I want it to get Clonable objects instead.

Reply:
Actually we don't need cloning and extra stuff, we can just deepCopy the Item and push it in the itemsArray of the new Slide.
```js
slide.itemsArray = draft.slides[draft.currentSlide].itemsArray.map(currentItem => {
  const copyObj: Item = deepCopyFunction(currentItem);
  return copyObj;
});
```

Reply:
I did it like that and seems to be working.